src/typedefs.rs

```diff
@@ -395,6 +395,30 @@ macro_rules! construct_secret_key {
         /// # Security:
         /// - __**Avoid using**__ `unprotected_as_bytes()` whenever possible, as it breaks all protections
         /// that the type implements.
+        ///
+        /// - The trait `PartialEq<&'_ [u8]>` is implemented for this type so that users are not tempted
+        /// to call `unprotected_as_bytes` to compare this sensitive value to a byte string. The trait
+        /// is implemented in such a way that the comparison happens in constant time. Thus, users should
+        /// prefer `SecretType == &[u8]` over `SecretType.unprotected_as_bytes() == &[u8]`.
+        /// Examples are shown below. Although the `orion::hazardous::stream::chacha20::SecretKey`
+        /// type is used, this example applies to any fixed-size store of sensitive data.
+        /// ```rust
+        /// use orion::hazardous::stream::chacha20::SecretKey;
+        ///
+        /// // Initialize a secret key with 32 random bytes.
+        /// let key_bytes: [u8; 32] =
+        ///     [0x2f, 0xbd, 0x03, 0xeb, 0x36, 0x66, 0xa9, 0x8c,
+        ///      0x39, 0x7f, 0xe7, 0xc3, 0x53, 0x4b, 0x9f, 0x8e,
+        ///      0x17, 0x76, 0x5d, 0x96, 0xf4, 0x08, 0x6c, 0x8f,
+        ///      0x25, 0x98, 0x80, 0xd9, 0xb2, 0xa5, 0x56, 0xce];
+        /// let secret_key = SecretKey::from_slice(&key_bytes).unwrap();
```

Comment:
We want to encourage users not to store secrets in source and use strong keys. Please remove `key_bytes` and initialize `secret_key` with `let secret_key = SecretKey::generate();`.

Reply:
For the later `assert` you can use a `"Not random secret value".as_bytes()` and negate to `!=`.

Reply:
Do you want to include a negative example? One that demonstrates the comparison could return false? Using something like `"Not random secret value".as_bytes()`, I mean.

Reply:
Actually, I see what you meant now. I like that idea. Making the change.